modules/opts.nix

```diff
@@ -84,6 +84,6 @@ in
           ) optionsAttrs
         );
       in
-      mkIf (content != "") content;
+      mkIf (content != "") (mkOrder 2 content); # Move options to top of file below global table
```

Comment:
Rather than using order priorities `1` and `2`, it's probably better to spread them out a little...

The "default" priority is 1000, with 500 being "before" and 1500 being "after".

https://github.com/NixOS/nixpkgs/blob/0141f779e4d05fb114557542dad8e13d5c261c75/lib/modules.nix#L1047-L1049

The first question we should ask is should we be before or after `mkBefore` (500)? Perhaps `M = {}` should be lower than 500, but maybe opts should be higher?

The second question is how much space should we leave around our order priority values for users to position their own definitions. Perhaps sticking to round numbers, such as multiples of 20, 20, 50, etc?

In theory, a user may wish to be able to add some lines _before_ one of our definitions, but after the other. We should leave enough room for them to have some flexibility.

Reply:
Were you thinking like 200/300 or 300/400 type of a assignment?

My intention originally was that we should have these global definitions definitely at the top because everything else needs them in scope and defined, as soon as possible. But, I could see a usage of weird `extraConfigLuaPre` injection and that would give them more flexibility with stacking usages.

Reply:
I think anyone using `mkOrder` in their config should hopefully be knowledgeable enough to check what priority the thing they want to be relateve to is. Someone using `mkBefore` may not care as much, but will probably expect their definition to end up before pretty much all others.

I'd suggest that `local _M = {}` should have a fairly low priority. 100 or 200 seems fine. However I'm not convinced the opt definitions should trump `mkBefore`, so maybe something more like 600 or 700 there?

What do you think?

Reply:
In my head I wanted to ensure these both were at the very top because they seem foundational. I'm not sure what use case someone would have for trying to mkBefore the opts themselves, especially if just from a mkBefore. To me, it felt like it would lead to unexpected results to have a user injected lua config automatically jump above variables they might try and access.

Reply:
That argument does make sense. To clarify my perspective:
- Users _already_ using `mkBefore` may be tripped up by this change
- Users may wish to define their own variables that are referenced from opts definitions
- I think it is less surprising for definitions wrapped using `mkBefore` to end up before our opts

Not saying I have a strong opinion though; I'm still somewhat on the fence.

Reply:
cc @GaetanLepage @traxys